agent/utils/ephemeral_ports.go

```diff
@@ -214,3 +214,32 @@ func getHostPortRange(numberOfPorts, start, end int, protocol string) (string, i
 
 	return fmt.Sprintf("%d-%d", resultStartPort, resultEndPort), resultEndPort, nil
 }
+
+// PortIsInRange returns true if the given port is within the start-end port range;
+// otherwise, returns false.
+func PortIsInRange(port, start, end int) bool {
+	if (port >= start) && (port <= end) {
+		return true
+	}
+	return false
+}
+
+// VerifyPortsWithinRange returns true if the actualPortRange is within the expectedPortRange;
+// otherwise, returns false.
+func VerifyPortsWithinRange(actualPortRange, expectedPortRange string) bool {
```

Comment:
QQ. Is this method used anywhere other than in task_test.go ? If not, could you move it to the test file?

Reply:
Also do we verify that a port is within range in our taskdef validation or elsewhere?  Where is this same validation done outside of our tests?

Reply:
> QQ. Is this method used anywhere other than in task_test.go ? If not, could you move it to the test file?

> Also do we verify that a port is within range in our taskdef validation or elsewhere? Where is this same validation done outside of our tests?

This method is used in both `agent/utils/ephemeral_ports_test.go` and `agent/api/task/task_test.go` in this PR. As it's using in different packages, I have to make it as a public method. 

And as fierlion@ suggested, we should validate a host port or a host port range assigned by ECS Agent after calling `getHostPort()` or `getHostPortRange()`. Therefore, I would like to keep `VerifyPortsWithinRange()` in  `agent/utils/ephemeral_ports.go`, and have a follow up PR to implement host port validations outside of our tests.

Reply:
Changes are made in [ [DHPA] Validate the host port/host port range found by ECS Agent before returning it #40](https://github.com/chienhanlin/amazon-ecs-agent/pull/40). Will create a PR when part 1 and part 2 are good to go.